packages/taco/src/taco.ts

```diff
@@ -92,6 +108,23 @@ export const decrypt = async (
 
 export const isAuthorized = async (
   provider: ethers.providers.Provider,
+  domain: Domain,
   messageKit: ThresholdMessageKit,
   ritualId: number,
-) => DkgCoordinatorAgent.isEncryptionAuthorized(provider, ritualId, messageKit);
+) => DkgCoordinatorAgent.isEncryptionAuthorized(provider, domain, ritualId, messageKit);
+
+export const registerEncrypters = async (
```

Comment:
This is very specific to the GlobalAllowList. I think we need to be careful here. Rituals may use the GlobalAllowList but I don't believe they have to. We'd need to first check here whether the `accessController` for the associated ritual is the global allow list contract before registering the encrypter. It doesn't have to be, although more often than not, it may be. 

cc @KPrasch

Reply:
Yes, the encryptor onboarding is a bit more complicated and it depends on https://github.com/nucypher/nucypher-ts/pull/306

We could hide this method from the `taco` API until we sort it out.

Reply:
👍 I think that would be prudent.